src/fiber/channel.rs

```diff
@@ -4929,6 +5220,28 @@ mod tests {
             .expect("accept channel success");
     }
 
+    #[tokio::test]
+    async fn test_create_public_channel() {
+        tracing_subscriber::fmt::fmt()
+            .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+            .pretty()
+            .init();
+
+        let _span = tracing::info_span!("node", node = "test").entered();
+
+        let node_a_funding_amount = 100000000000;
+        let node_b_funding_amount = 6200000000;
+
+        let (node_a, node_b, new_channel_id) = create_nodes_with_established_channel(
+            node_a_funding_amount,
+            node_b_funding_amount,
+            true,
+        )
+        .await;
+        // Wait for the channel announcement to be broadcasted
+        tokio::time::sleep(tokio::time::Duration::from_millis(5000)).await;
```

Comment:
seems need to add assertion here.

Reply:
Yes. I was trying to add an expected event so that we can be sure this is working. As I hinted on the discord server, I think it would be better if we can just see all the messages sent to the network actor. I had some problem implementing some kind of introspection functionality. I can add a `NetworkServiceEvent` for `ChannelAnnouncement` here, but I found `NetworkServiceEvent` had became somewhat bloated.